pkg/domain/ingress.go

```diff
@@ -0,0 +1,78 @@
+package domain
+
+type SiteIngress interface {
+	GetName() string
+	GetHost() string
+	GetPort() int
+	GetTarget() Port
```

Comment:
This is related with that idea I proposed sometime ago. If we define just platform specific structures, it won't
be easy to create generic methods (or even tests) that could use the generic representation of a given entity.

These interfaces are meant to represent generic types, so that we could have platform specific implementations
adding extra state/behavior, while still reusing methods that just need the generic part of those entities.